presto-main/src/main/java/com/facebook/presto/cost/PlanNodeStatsEstimateMath.java

```diff
@@ -118,27 +119,36 @@ public static PlanNodeStatsEstimate addStatsAndCollapseDistinctValues(PlanNodeSt
 
     private static PlanNodeStatsEstimate addStats(PlanNodeStatsEstimate left, PlanNodeStatsEstimate right, RangeAdditionStrategy strategy)
```

Comment:
We should remove `strategy` from this method and call it as we talked: `addNonOverlapping` (or similarly). This can be a separate PR though. 

`addAndSumDistinctValues` and `addAndCollapseDistinctValues` should just be `addNonOverlapping`.
While `addAndMaxDistinctValues` should just be `addOverlapping`.

I could do that if you don't mind.

Reply:
I don't mind. But let's do this as a separate PR.